Disable coloring if NO_COLOR is present in environ

antifob opened the pull request:
Disable coloring if `NO_COLOR` is present in environment variables.

rhelmot commented:
Please change the default to the empty string instead of None in order to properly conform to the no-color spec (passing NO_COLOR= should enable colors). You may also want to invert and rename the variable to only have a single negative (nocolor instead of no_nocolor).

antifob commented:
You're right, I misread the spec.
Updated the check, but the double-negation is pretty much unavoidable. I moved it to the code rather than the varname.

rhelmot commented:
thank you :)
